src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  fromFormat.forEach((format, index) => {
+    dateMap[format] = dateParts[index];
+  });
+
+  if (dateMap['YY']) {
+    const year = parseInt(dateMap['YY'], 10);
+
+    dateMap['YYYY'] = year < 30 ? '20' + dateMap['YY'] : '19' + dateMap['YY'];
+  }
+
+  if (dateMap['YYYY']) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
+  }
+
+  const newDateParts = toFormat.map((format) => dateMap[format]);
+
+  return newDateParts.join(toSeparator);
```

Comment:
The function does not handle the case where 'YY' is in `toFormat` and 'YYYY' is not in `fromFormat`. This would result in an undefined value for 'YY' when constructing `newDateParts`. You need to ensure that 'YY' is properly derived from 'YYYY' when necessary.